packages/wagmi/src/connectors/EIP6963Connector.ts

```diff
@@ -74,10 +74,18 @@ export class EIP6963Connector extends InjectedConnector {
       if (!eip6963Wallet || connectedEIP6963Rdns !== eip6963Wallet.info.rdns) {
         return true
       }
+    }
+
+    if (
+      (eip6963Wallet && eip6963Wallet.provider._state?.isUnlocked === undefined) ||
+      (eip6963Wallet && eip6963Wallet.provider._state?.isUnlocked)
```

Comment:
wouldn't it be the same to check 
if (eip6963Wallet && eip6963Wallet.provider?._state?.isUnlocked === true) { return true }